packages/vite/src/node/plugins/manifest.ts

```diff
@@ -42,6 +46,15 @@ export function manifestPlugin(config: ResolvedConfig): Plugin {
         }
       }
 
+      function replaceFileWithAsset(chunk: OutputChunk) {
+        const name = chunk.facadeModuleId ? getChunkName(chunk) : chunk.name
+        const asset = Object.values(bundle).find(
+          (otherChunk) => otherChunk.type === 'asset' && otherChunk.name == name
+        )
+        if (asset) chunk.fileName = asset.fileName
+        return chunk
+      }
```

Comment:
Didn't want to optimize it until we discuss the approach.

We should use an approach similar to `chunkToEmittedAssetsMap`, where we index the corresponding asset as we process the chunks, so that we can efficiently retrieve them here.

An alternative is to simply replace the `fileName` in the chunk beforehand, but I'm not sure if the processing order would allow that.

Reply:
Since multiple chunks can have the same name, this can result in wrong mapping if two entry assets have the same name. e.g. `logo.svg` and `logo.png`

Reply:
In this case we would be comparing resolved names, which include the file extension, as assets always include the file extension in their name, correct?